bool new_data
bool request
int8 shape_id
string shape_name